### PR TITLE
fix varies bugs

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
1.  fix the get child resource go loop variable change bug
2. add GenerationChangedPredicate to avoid tight loop on version 1.16 
3. fix the logging rollover bug by creating a separate log for each reconcile logic
4. replace status update to patch
5. refactoring and fix test
6. Add the local flight tracker as an example

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -48,14 +48,13 @@ jobs:
         pathInArchive: linux-amd64/helm
 
     - name: Setup Kind Cluster
-      uses: engineerd/setup-kind@v0.3.0
-      with:
-        version: "v0.7.0"
+      uses: helm/kind-action@v1.0.0-rc.1
```
Comment (wonderflow) [inline .github/workflows/go.yml]: I have fixed

File: pkg/controller/core/workloads/containerizedworkload/containerizedworkload_controller_helper.go [hunk 2]
```diff
@@ -32,6 +32,8 @@ func (r *Reconciler) renderDeployment(ctx context.Context,
 	}
 	// the translator lib doesn't set the namespace
 	deploy.Namespace = workload.Namespace
+	// make sure we don't have opinion on the replica count
+	deploy.Spec.Replicas = nil
```
Comment (wonderflow) [inline pkg/controller/core/workloads/containerizedworkload/containerizedworkload_controller_helper.go:36]: What if we don't use ManualScaler for an AppConfig?

**Review by wonderflow — COMMENTED** [review]
I see flight-tracker was added as an example, I'm not sure if it's proper to put in this repo. Can you give some guide in README?

File: pkg/controller/core/workloads/containerizedworkload/containerizedworkload_controller_helper.go (hunk 2 above)
Comment (ryanzhang-oss) [reply pkg/controller/core/workloads/containerizedworkload/containerizedworkload_controller_helper.go:36]: do we need to have an opinion on this field in other cases?

**Comment by ryanzhang-oss** [comment]
> I see flight-tracker was added as an example, I'm not sure if it's proper to put in this repo. Can you give some guide in README?

I wonder if there is any concerns on using it as an example? I am actually planning to add a test based on it too.

**Review by wonderflow — APPROVED** [review]

**Comment by wonderflow** [comment]
> > I see flight-tracker was added as an example, I'm not sure if it's proper to put in this repo. Can you give some guide in README?
> 
> I wonder if there is any concerns on using it as an example? I am actually planning to add a test based on it too.

I just wonder if people will get confused for this example here.

**Comment by wonderflow** [comment]
I think it's fine if just for test, let me merge this PR.

**Pull request merged**